ckanext/advancedauth/auth.py

```diff
@@ -120,6 +126,15 @@ def advancedauth_wrapper_function(next_func, context, data_dict=None):
     if disallow_anonymous_access and func_name not in action_allowlist:
         advancedauth_check_access(next_func, context, data_dict)
 
+    # if user is sysadmin, skip all checks
+    username = context.get("user", "")
+    user = _get_user(username)
```

Comment:
You're sure the fetch is necessary and not already existing in `user_obj` or `auth_user_obj`?

Reply:
@rjruizes Okay I did some more digging. So we can retrieve `auth_user_obj` anywhere on the normal CKAN pages, but within the audit explorer app this generates an error because the context looks like the following as an example. 
```
 {
	'model': <module 'ckan.model' from '/srv/app/src/ckan/ckan/model/__init__.py'>
	'auth_user_obj': None
	'__auth_user_obj_checked': True
	'session': <sqlalchemy.orm.scoping.scoped_session object at 0x4014c03a60>
	'user': 'sysadmin'
 }
```
This may be because in the routes, the context is manually created like the following:
```
@audit_table.route("/getresources")
def list_resources():
    if toolkit.g.userobj and toolkit.g.userobj.sysadmin:
        packages = current_package_list_with_resources(
            {"user": toolkit.g.userobj.name, "model": model}, {}
        )
        /* rest of code */
```

I can try passing in the full context so that we can remove the fetch operation, though I'm not sure if this is possible. I'm not very familiar with flask. Is there a reason we shouldn't?

Reply:
Thanks for looking at this. Sounds like we're creating an incomplete context- probably in a few places that I've written. I think it'd be better to fix that, i.e. do CKAN requests more like CKAN does. But if it's too much effort, the extra request works.

Reply:
I found in the documentation for toolkit.get_action:
```
If a context of None is passed to the action function then the default context dict will be created.
```
So the fix was simple enough I went ahead and replaced the ckan actions with toolkit.get_action and passed None for context. I removed the fetch from db for user objects. Integration tool, searchterms, and summarystats still work correctly so no changes are needed there I think.